src/graph/query_graph.c

```diff
@@ -145,22 +145,14 @@ void _BuildQueryGraphAddEdge(const GraphContext *gc,
     Node *src = QueryGraph_GetNodeByAlias(qg, src_node->alias);
     Node *dest = QueryGraph_GetNodeByAlias(qg, dest_node->alias);
     Edge *e = Edge_New(src, dest, edge->ge.label, edge->ge.alias);
-
-    // Get relation matrix.
+    
+    //Set edge relation ID.
     if(edge->ge.label == NULL) {
         Edge_SetRelationID(e, GRAPH_NO_RELATION);
-        e->mat = Graph_GetAdjacencyMatrix(g);
     } else {
         Schema *s = GraphContext_GetSchema(gc, edge->ge.label, SCHEMA_EDGE);
-        if(s) {
-            Edge_SetRelationID(e, s->id);
-            e->mat = Graph_GetRelationMatrix(g, s->id);
-        }
-        else {
-            /* Use a zeroed matrix.
-             * TODO: either use a static zero matrix, or free this one. */
-            GrB_Matrix_new(&e->mat, GrB_BOOL, Graph_NodeCount(g), Graph_NodeCount(g));
-        }
+        if(s) Edge_SetRelationID(e, s->id);
+        else Edge_SetRelationID(e, GRAPH_UNKNOWN_RELATION);
```

Comment:
This would only be the case if the user specified a non-existent relationship, right? Can you add a comment to clarify what's being handled here?

Reply:
That's the case.